chat-widget/src/components/livechatwidget/common/PersistentConversationHandler.ts

```diff
@@ -0,0 +1,252 @@
+    private processHistoryMessages(messagesDescOrder: any[]) {
+        for (const message of messagesDescOrder) {
+            try {
+                const activity = this.processMessageToActivity(message);
+
+                if (activity) {
+                    dispatchCustomEvent(ChatWidgetEvents.ADD_ACTIVITY, { activity });
+                    const dividerActivity = this.createDividerActivity(activity);
+
+                    if (dividerActivity) {
+                        dispatchCustomEvent(ChatWidgetEvents.ADD_ACTIVITY, { activity: dividerActivity });
+                    }
+                    this.lastMessage = activity;
+                }
+
+            } catch (error) {
+                TelemetryHelper.logActionEvent(LogLevel.ERROR, {
+                    Event: TelemetryEvent.ConvertPersistentChatHistoryMessageToActivityFailed,
+                    ExceptionDetails: error,
+                });
+            }
+        }
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    private async fetchHistoryMessages(): Promise<any[]> {
+
+        if (!this.shouldPull()) {
+            return [];
+        }
+
+        const options = {
+            pageSize: this.pageSize,
+            pageToken: this.pageToken || undefined,
+        };
+
+        try {
+            const response = await this.facadeChatSDK.fetchPersistentConversationHistory(options);
+            const { chatMessages: messages, nextPageToken: pageToken } = response;
+            this.pageToken = pageToken || null;
+
+            if (pageToken === null) {
+                this.isLastPull = true;
+                // Dispatch event when we reach the end of available history
+                dispatchCustomEvent(ChatWidgetEvents.NO_MORE_HISTORY_AVAILABLE);
+            }
+
+            // if chatMessages is null, return empty array
+            if (!messages) {
+                return [];
+            }
+            return messages;
+        } catch (error) {
+            TelemetryHelper.logSDKEvent(LogLevel.ERROR, {
+                Event: TelemetryEvent.FetchPersistentChatHistoryFailed,
+                ExceptionDetails: error,
+            });
+
+            this.isLastPull = true;
+            this.pageToken = null;
+            // Dispatch event when there's an error to stop loading banner
+            dispatchCustomEvent(ChatWidgetEvents.NO_MORE_HISTORY_AVAILABLE);
+            return [];
+        }
+    }
+
+    private shouldPull(): boolean {
+        return !this.isLastPull;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    private processMessageToActivity(message: any): any {
+        try {
+            const activity = convertPersistentChatHistoryMessageToActivity(message);
+
+            activity.id = activity.id || `activity-${this.count}`;
+            activity.channelData = {
+                ...activity.channelData,
+                metadata: {
+                    count: this.count,
```

Comment:
For my understanding, what does activity.channelData.metadata.count do again? It's not ordering right, that is done with sequenceId? Why is it important to add here?